core/src/Streamly/Internal/Data/ParserK/Type.hs

```diff
@@ -487,17 +491,24 @@ adaptWith
     -> Input a
     -> m (Step a m r)
 adaptWith pstep initial extract cont !relPos !usedCount !input = do
+    -- We ensure that this does not happen
+#ifdef DEBUG
+    when (relPos /= 0) $ error $ "adaptWith: entry relPos " ++ show relPos
+#endif
     res <- initial
     case res of
         ParserD.IPartial pst -> do
-            -- XXX can we come here with relPos 1?
             if relPos == 0
             then
                 case input of
-                    Chunk arr -> parseContChunk usedCount pst arr
+                    -- In element parser case chunk is just one element
+                    Chunk element -> parseContChunk usedCount pst element
                     None -> parseContNothing usedCount pst
             -- XXX Previous code was using Continue in this case
-            else pure $ Partial relPos (parseCont usedCount pst)
+            else
+                -- We consumed previous input, need to fetch the next
+                -- input from the driver.
+                pure $ Partial relPos (parseCont usedCount pst)
```

Comment:
Is this a statement or is there an issue or question here?

This is ParserK.Partial - it moves as per position index. Positive means move forward, negative means move backward. This code has not changed and it seems correct. If relPos is 1 we move to the next element, because index 1 is the end in this parser.